debezium-core/src/test/java/io/debezium/connector/base/ChangeEventQueueShutdownTest.java

```diff
@@ -0,0 +1,204 @@
+ * Test for the queue shutdown mechanism that prevents coordinator thread memory leaks.
+ *
+ * This test validates the fix for the issue where coordinator threads get stuck
+ * in infinite blocking loops when the queue is full and the consumer thread dies.
+ *
+ * @author Yashi Srivastava
+ */
+public class ChangeEventQueueShutdownTest {
+
+    /**
+     * Test that shutdown mechanism prevents infinite blocking in doEnqueue
+     */
+    @Test
+    public void shouldUnblockThreadsOnShutdown() throws Exception {
+        // Create a small queue to make it easy to fill
+        int queueSize = 5;
+        ChangeEventQueue<DataChangeEvent> queue = new ChangeEventQueue.Builder<DataChangeEvent>()
+                .maxBatchSize(3)
+                .maxQueueSize(queueSize)
+                .loggingContextSupplier(() -> LoggingContext.forConnector("test", "test", "test"))
+                .pollInterval(Duration.ofMillis(100))
+                .build();
+
+        // Fill the queue to capacity
+        DataChangeEvent testEvent = new DataChangeEvent(null);
+        for (int i = 0; i < queueSize; i++) {
+            queue.doEnqueue(testEvent);
+        }
+
+        // Create thread that will block trying to enqueue
+        AtomicBoolean enqueueCompleted = new AtomicBoolean(false);
+        AtomicReference<Exception> enqueueException = new AtomicReference<>();
+        CountDownLatch enqueueStarted = new CountDownLatch(1);
+        CountDownLatch enqueueFinished = new CountDownLatch(1);
+
+        Thread enqueueThread = new Thread(() -> {
+            try {
+                enqueueStarted.countDown();
+                // This should block because queue is full
+                queue.doEnqueue(testEvent);
+                enqueueCompleted.set(true);
+            }
+            catch (InterruptedException e) {
+                enqueueException.set(e);
+            }
+            catch (Exception e) {
+                enqueueException.set(e);
+            }
+            finally {
+                enqueueFinished.countDown();
+            }
+        });
+
+        enqueueThread.start();
+
+        // Wait for thread to start and get blocked
+        assertTrue("Enqueue thread should start", enqueueStarted.await(5, TimeUnit.SECONDS));
+        Thread.sleep(200); // Give time to reach blocking condition
+
+        // Verify thread is alive and blocked
+        assertTrue("Thread should be alive and blocked", enqueueThread.isAlive());
+
+        // Shutdown queue - should unblock the thread
+        long shutdownStart = System.currentTimeMillis();
+        queue.shutdown();
+
+        // Wait for thread to finish
+        boolean threadFinished = enqueueFinished.await(5, TimeUnit.SECONDS);
+        long totalDuration = System.currentTimeMillis() - shutdownStart;
+
+        // Verify shutdown worked
+        assertTrue("Thread should finish after shutdown", threadFinished);
+        assertTrue("Shutdown should be quick (under 2 seconds)", totalDuration < 2000);
```

Comment:
What's rationale for this ? It will depend on pollInterval right ? according to line 277 in ChangeEventQueue